### PR TITLE
Added DAO tests.

### PR DESCRIPTION
**Pull request by TheSprtCz**
# Overview
* Added DAO tests for every DAO class.
  * Some tests have combined create and remove tests because rollback didn't work when it had to also remove rows from additional tables, like roles or degrees.
* Added testing migrations.
  * Also added testing configuration written in markdown (Github Flavored Markdown) in DB.md file.

This PR is dependent on #28

### CODE REVIEW

File: studenthub-portal/pom.xml [hunk 1]
```diff
@@ -56,6 +56,10 @@
       <artifactId>dropwizard-assets</artifactId>
     </dependency>
     <dependency>
+      <groupId>io.dropwizard</groupId>
+      <artifactId>dropwizard-testing</artifactId>
+    </dependency>
```
Comment (sbunciak) [inline studenthub-portal/pom.xml]: Shouldn't it have a scope of "testing"?

File: studenthub-portal/src/test/java/cz/studenthub/db/AbstractDAOTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    SessionFactory sessionFactory = database.getSessionFactory();
+    Session session = sessionFactory.openSession();
+    Transaction transaction = session.beginTransaction();
+    session.doWork(connection -> {
+      Database database = null;
+      try {
+        database = DatabaseFactory.getInstance() 
+            .findCorrectDatabaseImplementation(new JdbcConnection(connection));
+      } catch (DatabaseException e1) {
+        // TODO Auto-generated catch block
```
Comment (sbunciak) [inline studenthub-portal/src/test/java/cz/studenthub/db/AbstractDAOTest.java]: Remove this (and similar) comment, consider using proper logging.

File: studenthub-portal/src/test/java/cz/studenthub/db/CompanyDAOTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.List;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import cz.studenthub.core.Company;
+import cz.studenthub.core.CompanyPlan;
+import cz.studenthub.core.CompanySize;
+import cz.studenthub.core.Country;
+
+public class CompanyDAOTest extends AbstractDAOTest{
```
Comment (sbunciak) [inline studenthub-portal/src/test/java/cz/studenthub/db/CompanyDAOTest.java]: Please run source code formatter for all .java files (there is no space between 'AbstractDAOTest' and '{'
This problem occurs in other files as well.

File: studenthub-portal/src/test/java/cz/studenthub/db/CompanyDAOTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    @Test
+    public void createCompany() {
+    	Company uni = new Company("New", "www.nothing.eu", "Liberec", Country.CZ, "www.nothing.eu/logo.png", CompanySize.SMALL,
+          CompanyPlan.TIER_2);
+      inRollbackTransaction(() -> {
+        Company created =  companyDAO.createOrUpdate(uni);
+        List<Company> companies = companyDAO.findAll();
+        assertNotNull(created.getId());
+        assertEquals(9, companies.size());
+        assertEquals("New", created.getName());
```
Comment (sbunciak) [inline studenthub-portal/src/test/java/cz/studenthub/db/CompanyDAOTest.java]: why not asserting object equality?

File: studenthub-portal/src/test/java/cz/studenthub/db/TopicDAOTest.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+
+import cz.studenthub.core.Company;
+import cz.studenthub.core.Topic;
+import cz.studenthub.core.TopicDegree;
+import cz.studenthub.core.User;
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
```
Comment (sbunciak) [inline studenthub-portal/src/test/java/cz/studenthub/db/TopicDAOTest.java]: why is this necessary?

File: studenthub-portal/src/test/resources/migrations-test.xml [hunk 6]
```diff
@@ -0,0 +1,1118 @@
...
+        <createSequence catalogName="STUDENTHUB" schemaName="PUBLIC" sequenceName="TOPICS_ID_SEQ"/>
+    </changeSet>
+    <changeSet author="phala (generated)" id="1489653927422-5" objectQuotingStrategy="QUOTE_ALL_OBJECTS">
+        <createSequence catalogName="STUDENTHUB" schemaName="PUBLIC" sequenceName="UNIVERSITIES_ID_SEQ"/>
+    </changeSet>
+    <changeSet author="phala (generated)" id="1489653927422-6" objectQuotingStrategy="QUOTE_ALL_OBJECTS">
+        <createSequence catalogName="STUDENTHUB" schemaName="PUBLIC" sequenceName="USERS_ID_SEQ"/>
+    </changeSet>
+    <changeSet author="phala (generated)" id="1489653927422-7" objectQuotingStrategy="QUOTE_ALL_OBJECTS">
+        <createTable catalogName="STUDENTHUB" schemaName="PUBLIC" tableName="COMPANIES">
```
Comment (sbunciak) [inline studenthub-portal/src/test/resources/migrations-test.xml]: why populating the schema? shouldn't we use the "production" schema in tests?

File: studenthub-portal/src/test/resources/DB.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+| 4   | Company Four                       | c4.com/picture.jpg     | Brno       | CZ      | www.c4.com     | STARTUP   | TIER_1  |
+| 5   | Company Five                       | c5.com/us.jpg          | Prague     | CZ      | www.c5.com     | MEDIUM    | TIER_3  |
+| 6   | Company Six                        | c6.com/us.gif          | Olomouc    | CZ      | www.c6.com     | CORPORATE | TIER_2  |
+| 7   | Company Seven                      | c7.com/img.png         | Trencin    | SK      | www.c7.com     | MEDIUM    | TIER_3  |
+| 8   | Company Eight                      | c8.com/logo.jpg        | Myjava     | Sk      | www.c8.com     | STARTUP   | TIER_1  |
+
+## Users
+
+(Password field is not present because it is encoded and irrelevant when testing, password encoding test does not need DB)
+| id  | username           | email                       | name               | phone        | lastLogin      | company | faculty | roles                        | tags                     |
```
Comment (sbunciak) [inline studenthub-portal/src/test/resources/DB.md]: this table is not rendered properly on Github

File: studenthub-portal/src/test/resources/DB.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+## Topics
+
+| id  | title               | shortAbstract                       | description                | leader | creator | supervisors      | degrees                  | tags                      |
+| --- | ------------------- | ----------------------------------- | -------------------------- | ------ | ------- | ---------------- | ------------------------ | ------------------------- |
+| 1   | Melon cutter        | Create simple laser melon cutter    | Laser melon cutter         | 8      | 9       | [3,4,6]          | [PhD,MASTER]             | [Laser, Assembler, C]     |
+| 2   | Dropwizard          | Simple app using Dropwizard stack   | REST endpoints             | 11     | 11      | [7,5,2]          | [BACHELOR, HIGH_SCHOOL]  | [Java, REST, Web]         |
+| 3   | Eclipse plugin      | Create custom eclipse plugin        | Eclipse                    | 10     | 10      | [5,7,4]          | [MASTER, BACHELOR]       | [Java, C, Maven]          |
+| 4   | React UI            | Create nice and functional UI       | JavaScript & React         | 9      | 9       | [6,3,2]          | [HIGH_SCHOOL,PhD]        | [JavaScript, Web]         |
+
+## Topic Application
```
Comment (sbunciak) [inline studenthub-portal/src/test/resources/DB.md]: this table is not rendered properly on Github

**Comment by sbunciak** [comment]
@TheSprtCz please remove "Added named query for every ManyToOne relation." commit from this PR so that it can be merged.

File: studenthub-portal/pom.xml [hunk 9]
```diff
@@ -164,6 +169,15 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+	      <groupId>org.apache.maven.plugins</groupId>
+	      <artifactId>maven-surefire-plugin</artifactId>
+	      <configuration>
+	        <classpathDependencyExcludes>
+	          <classpathDependencyExclude>cz.studenthub:studenthub-ui</classpathDependencyExclude>
```
Comment (sbunciak) [inline studenthub-portal/pom.xml]: what's the benefit in not having this jar on classpath?

Comment (TheSprtCz) [inline studenthub-portal/pom.xml]: I am still unable to run "mvn clean test" because of swagger doc generation on classpath, but I can run it in student-portal, so I excluded it just from studenthub-portal.
